### DISCUSSION TITLE
Avatar group `stacking` prop

### DISCUSSION THREAD

**Discussion by isBatak** [chakra-ui/chakra-ui#8436] [Ideas] [answered]
### Description

Add support or stacking prop that will control stacking context of avatar group.

### Problem Statement/Justification

It's most common for avatar group designs to have stacking set to `last-on-top`, but Chakra for some reason has it set to `first-on-top`. We need an option to change it per component. 

Figma has this property:
![image](https://github.com/chakra-ui/chakra-ui/assets/4128883/5568f155-befa-4ebf-85ae-b0a541f6457d)

### Proposed Solution or API

```tsx
<AvatarGroup size='md' max={2} stacking="first-on-top">
  <Avatar name='Ryan Florence' src='https://bit.ly/ryan-florence' />
  <Avatar name='Segun Adebayo' src='https://bit.ly/sage-adebayo' />
  <Avatar name='Kent Dodds' src='https://bit.ly/kent-c-dodds' />
  <Avatar name='Prosper Otemuyiwa' src='https://bit.ly/prosper-baba' />
  <Avatar name='Christian Nwamba' src='https://bit.ly/code-beast' />
</AvatarGroup>
```

**Comment by Omikorin** [comment]
I think it's a good suggestion

**Comment by kwaimind** [comment]
+1 on this issue. I spent most of this morning trying to work out how Chakra does this by default to see if I can add an inverted option to my theme but my only solution was using a `zIndex` on each of the avatars and then a final `zIndex` in the theme for the `excess`. A `stacking` prop would be much appreciated. Happy to help if needed.

However, I think you got your order wrong in the description. Chakra uses `first-on-top` by default right and we would want to add a `last-on-top` option?
<img width="183" alt="image" src="https://github.com/chakra-ui/chakra-ui/assets/4730761/1496f66e-485f-481c-a969-9134e85edc4e">

**Comment by abolfazlcodes** [comment]
Hey,
can i work on this?

**Comment by FezVrasta** [comment]
The way I resolved it is with this configuration:

```
const baseStyle = definePartsStyle({
  container: {
    ...Array.from({ length: 20 }, (_, i) => ({
      [`&:nth-last-child(${i})`]: {
        zIndex: i,
      },
    })).reduce((acc, val) => ({ ...acc, ...val }), {}),
  },
  excessLabel: {
    zIndex: 20,
  },
});

export const avatarTheme = defineMultiStyleConfig({ baseStyle });
```

You can update it to support more elements if needed, but I think 20 is a good number.

![CleanShot 2024-05-09 at 2  57 05@2x](https://github.com/chakra-ui/chakra-ui/assets/5382443/2821da07-7f4c-47d5-82df-ff6cce529eec)

**Comment by segunadebayo** [comment, marked as answer]
AvatarGroup stacking shipped. Use `stacking="last-on-top"`.
